code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceAllocate.java

```diff
@@ -30,6 +51,43 @@ public InstanceAllocate() {
     }
 
     @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        // Check resourceType for instance and whether the host is a cluster or not.
+        // If resourceType is a cluster (host), do not bother with normal allocation process.
+        final Instance instance = (Instance)state.getResource();
+
+        // check requestedHostId to see whether it's a cluster or not
+        Long requestedHostId = DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_REQUESTED_HOST_ID).as(Long.class);
+        if (requestedHostId != null) {
+            Host potentialCluster = objectManager.loadResource(Host.class, requestedHostId);
+            if (objectManager.isKind(potentialCluster, ClusterConstants.KIND)) {
+                log.info("Instance [{}] requested for a cluster: [{}]", instance, potentialCluster);
+
+                // create instanceHostMap binding with cluster.
+                objectManager.create(InstanceHostMap.class,
+                        INSTANCE_HOST_MAP.HOST_ID, requestedHostId,
+                        INSTANCE_HOST_MAP.INSTANCE_ID, instance.getId());
+
+                // update nic with active subnet
+                Nic nic = nicDao.getPrimaryNic(instance);
+                Network network = objectManager.loadResource(Network.class, nic.getNetworkId());
+                if (network != null) {
```

Comment:
I think we should still be good here. This code was actually pulled from NetworkAllocationCandidates.  NetworkAllocationCandidates creates a bunch of AllocationCandidates with the different subnets available.  But ultimately, I believe the instance allocator just picks the first candidate that fits all the different criterias (the network constraint being one of the criteria)